src/stats.c

```diff
@@ -33,6 +34,69 @@ static struct {
     double user;
 } used_cpu;
 
+// log slow command
+#define BUILD_CMD_ITEM(cmd, type, access) #cmd,
+static const char * cmd_table[] = {CMD_DO(BUILD_CMD_ITEM)};
+static const size_t CMD_NUM = sizeof(cmd_table) / sizeof(char*);
+static uint32_t *counts_sum;
+
+static struct dict slow_counts; // node dsn => counts
+static pthread_mutex_t counts_mutex;
+
+
+static int stats_init_slow_log()
+{
+    int err;
+
+    dict_init(&slow_counts);
+    counts_sum = cv_malloc(CMD_NUM * sizeof(uint32_t));
+
+    if ((err = pthread_mutex_init(&counts_mutex, NULL)) != 0) {
+        LOG(ERROR, "pthread_mutex_init: %s", strerror(err));
+        return CORVUS_ERR;
+    }
+
+    char dsn[ADDRESS_LEN];
+    for (size_t i = 0; i != config.node.len; i++) {
+        struct address *addr = config.node.addr + i;
+        snprintf(dsn, sizeof(dsn), "%s:%d", addr->ip, addr->port);
+        const char *clone = cv_strdup(dsn);
+        uint32_t *counts = cv_calloc(CMD_NUM, sizeof(uint32_t));
+        dict_set(&slow_counts, clone, counts);
+    }
+
+    return CORVUS_OK;
+}
+
+static void stats_free_slow_log()
+{
+    struct dict_iter iter = DICT_ITER_INITIALIZER;
+    DICT_FOREACH(&slow_counts, &iter) {
+        cv_free(iter.key);
+        cv_free(iter.value);
+    }
+
+    dict_free(&slow_counts);
+    cv_free(counts_sum);
+    pthread_mutex_destroy(&counts_mutex);
+}
+
+void stats_log_slow_cmd(struct command *cmd)
```

Comment:
Using lock in the main logic flow is so heavy. Have a look at `stats_node_info_agg` and try to find a simple lock free solution.